File: typescript-automation/src/dsl/protocols/channel.ts
export type Channel = "Web" | "API";

let resolved: Channel | undefined;

export function currentChannel(): Channel {
  if (resolved !== undefined) return resolved;
  const raw = process.env["CHANNEL"];
  if (!raw || raw.trim() === "") {
    throw new Error("Environment variable CHANNEL is required. Valid values: Web, API");
  }
  const trimmed = raw.trim();
  if (trimmed !== "Web" && trimmed !== "API") {
    throw new Error(`Unknown channel '${trimmed}'. Valid values: Web, API`);
  }
  resolved = trimmed;
  return resolved;
}

File: python-automation/src/automation/dsl/protocols/channel.py
import os
from enum import Enum
from typing import Optional


class Channel(Enum):
    WEB = "Web"
    API = "API"


_cached: Optional[Channel] = None


def current_channel() -> Channel:
    global _cached
    if _cached is None:
        _cached = _resolve()
    return _cached


def reset_cached_channel() -> None:
    global _cached
    _cached = None


def _resolve() -> Channel:
    raw = os.environ.get("CHANNEL", "").strip()
    if not raw:
        raise RuntimeError(
            "Environment variable CHANNEL is required. Valid values: Web, API"
        )
    try:
        return Channel[raw.upper()]
    except KeyError as e:
        raise RuntimeError(
            f"Unknown channel '{raw}'. Valid values: Web, API"
        ) from e
